### PR TITLE
MINOR: Fix error in testRestartBrokerNoErrorIfMissingPartitionMetadata

### PR DESCRIPTION
**Pull request by brandboat**
related to https://github.com/apache/kafka/pull/16165#issuecomment-2149821804

fix error in 3.7 backport https://issues.apache.org/jira/browse/KAFKA-16814 

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by brandboat** [comment]
gentle ping @soarez , @chia7712 , please take a look when you are available, many thanks 😃

**Review by soarez — APPROVED** [review]
Thank you @brandboat.

Tested locally, compilation and tests look good. Will wait for CI before merging.

**Comment by soarez** [comment]
Failed tests are unrelated. Merging

**Pull request merged**
